streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -523,67 +526,41 @@ public KafkaStreams(final Topology topology,
         this(topology.internalTopologyBuilder, config, clientSupplier);
     }
 
+    /**
+     *
+     * @param internalTopologyBuilder
+     * @param config
+     * @param clientSupplier
+     *
+     * @throws org.apache.kafka.streams.errors.ProcessorStateException if {@link StateDirectory} cannot be created
```

Comment:
I think we can be more specific about this error, e.g. sth like `.. cannot be created because the specified state directory either does not exist or cannot be accessed`.

Also I'm wondering if we should wrap this exception as a general StreamsException since it is user facing public API and `ProcessorStateException` was meant to used internally only.